clang/lib/Sema/SemaDecl.cpp

```diff
@@ -8341,6 +8341,11 @@ void Sema::CheckShadow(NamedDecl *D, NamedDecl *ShadowedDecl,
     // shadowing context, but that's just a false negative.
   }
 
+  // Skip shadowing check if we're in a class scope, dealing with an enum
+  // constant in a different context.
+  if (S->isClassScope() && isa<EnumConstantDecl>(D) &&
```

Comment:
Can we simply check that `NewDC->isRecord()` here instead of using a new `Scope` parameter?

Reply:
I think in this case `NewDC` refers to `EnumDeclaration`, so maybe checking `RedeclContext` would work, though it needs verification. Is relying on `Scope` not permitted here?

Reply:
> Is relying on Scope not permitted here?

I don't think there are any problems with that, at least none that I'm aware of. Yet it adds additional parameter to `Sema::CheckShadow` function which is checked only for one case with `EnumConstantDecl` but needs to be passed to each invocation of `Sema::CheckShadow`. It seems checking a correct `DeclContext` can help to achieve the same result without an additional parameter.